### PR TITLE
#159846051: add tests for login and improve overall test coverage

### PR DESCRIPTION
**Pull request by chidioguejiofor**
#### What does this PR do?
- adds tests for login route (POST /api/users/login)
- removes unnecessary route that gets a user GET (/users)
- improves overall test coverage
- adds validator to the login route

#### Description of Task to be completed?
- add test for login route
- improve overall test coverage

#### How should this be manually tested?
- run `npm test` from your console to see the new test coverage

#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- 159846051

#### Screenshots (if appropriate)
screen shot 2018-08-27 at 5 59 07 pm

#### Questions:
- None

### CODE REVIEW

File: server/controllers/UserController.js [hunk 1]
```diff
@@ -65,8 +65,7 @@ class UserController {
   static login(req, res, next) {
     return passport.authenticate(
       'local', { session: false },
-      (err, user, info) => {
-        if (err) return next(err);
```
Comment (megame24) [inline server/controllers/UserController.js]: Why did you remove this error handling?

File: server/controllers/UserController.js [hunk 2]
```diff
@@ -78,30 +77,17 @@ class UserController {
             user: userControllerHelper.userDetails(user, token),
           });
         }
-        return res.status(422)
-          .json(info);
+        const errorObj = {
+          errors: {
+            message: 'email and password combination not found',
```
Comment (megame24) [inline server/controllers/UserController.js]: Make use of the `errorHelper` function created to enforce consistency

File: server/routes/api/index.js [hunk 3]
```diff
@@ -12,5 +12,6 @@ router.use('/', userRouter);
 router.use('/', articleRouter);
 router.use('/', followRouter);
 router.use('/', commentRouter);
+router.use((err, req, res, next) => next(err));
```
Comment (megame24) [inline server/routes/api/index.js]: Why do you need this middleware?

File: server/controllers/UserController.js [hunk 4]
```diff
@@ -78,30 +77,17 @@ class UserController {
...
   }
 
-  /**
-   * Get the details of a logged in user
-   * @param {Object} req the request body
-   * @param {Object} res the response body
-   * @param {function} next a call to the next function
-   * @returns {Object} the response body
-   */
-  static get(req, res, next) {
```
Comment (megame24) [inline server/controllers/UserController.js:97]: Why did you delete this controller?

File: server/routes/api/users.js [hunk 5]
```diff
@@ -6,12 +6,6 @@ import AuthMiddleware from '../../utils/middleware/AuthMiddleware';
 
 const router = Router();
 
-router.get(
```
Comment (megame24) [inline server/routes/api/users.js:10]: Why did you delete this route?

File: server/test/routes/oauth.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable */
+import { describe, it } from 'mocha';
+import { expect, use, request } from 'chai';
+import chaiHttp from 'chai-http';
+import server from '../../index';
+use(chaiHttp);
+
+
+describe('Social Login', () => {
```
Comment (megame24) [inline server/test/routes/oauth.test.js]: I thought we agreed to remove these tests since they aren't really testing for anything, order than making a request to `Facebook, LinkedIn, and google`.

Comment (chidioguejiofor) [reply server/test/routes/oauth.test.js]: Okay.
I was thinking that it would drop the test coverage

File: server/routes/api/users.js (hunk 5 above)
Comment (chidioguejiofor) [reply server/routes/api/users.js:10]: Because it retrieve information of the user and that is already implemented in the get profile route

File: server/controllers/UserController.js (hunk 4 above)
Comment (chidioguejiofor) [reply server/controllers/UserController.js:97]: It does nothing

File: server/routes/api/users.js (hunk 5 above)
Comment (chidioguejiofor) [reply server/routes/api/users.js]: The get user route is not being used anyway in the project. It just retrieves the user and returns a user. This is part of what was reducing test coverage

File: server/controllers/UserController.js (hunk 4 above)
Comment (megame24) [reply server/controllers/UserController.js:97]: ok

File: server/routes/api/users.js (hunk 5 above)
Comment (megame24) [reply server/routes/api/users.js:10]: alright

File: server/test/routes/oauth.test.js (hunk 6 above)
Comment (megame24) [reply server/test/routes/oauth.test.js]: It won't, remove it and see for yourself, all it does is send a request to those social media sites. The tests aren't really testing any controller or function in our application.

**Review by megame24 — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]
